lib/puppet/functions/stdlib/nested_values.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+# @summary Get list of nested values from given hash
+#   This function will return list of nested Hash values and returns list of values in form of Array
+#
+# @example Example Usage:
+#   $hash = {
+#     "key1" => "value1",
+#     "key2" => { "key2.1" => "value2.1"},
+#     "key3" => "value3"
+#   }
+#   $data = $hash.stdlib::nested_values
+#   #Output : ["value1", "value2.1", "value3"]
+Puppet::Functions.create_function(:'stdlib::nested_values') do
+  # @param options A (nested) hash
+  # @return All the values found in the input hash included those deeply nested.
+  dispatch :nested_values do
+    param 'Hash', :options
```

Comment:
`options` is a bit of an odd name for this parameter??  Makes sense in the context of `deferrable_epp`, but without that context, not so much.  Maybe just call it `hash`???